File: 1710-maximum-units-on-a-truck/1710-maximum-units-on-a-truck.js
/**
 * @param {number[][]} boxTypes
 * @param {number} truckSize
 * @return {number}
 */
var maximumUnits = function(boxTypes, truckSize) {
  /*
    Sort boxes with greatest number of units (descending order)
    Decrement truckSize after taking on each box
    Return num of units
  */
  
  boxTypes.sort((a,b) => b[1] - a[1]);
  let totalUnits = 0, i = 0;
  
  for (let i = 0; i < boxTypes.length; i++) {
    const [ boxes, boxUnits ] = boxTypes[i];
    if (truckSize >= boxes) {
      totalUnits += boxes * boxUnits;
      truckSize -= boxes;
    } else if (truckSize > 0 && truckSize < boxes) {
      totalUnits += truckSize * boxUnits;
      truckSize = 0;
    } else {
      break;
    } 
  }
  
  return totalUnits;
};

File: 1710-maximum-units-on-a-truck/1710-maximum-units-on-a-truck.py
class Solution:
    def maximumUnits(self, boxTypes: List[List[int]], truckSize: int) -> int:
      boxTypes.sort(key = lambda x: x[1])
      boxTypes.reverse()  
      total = 0
      
      for package in boxTypes:
        boxes, units = package[0], package[1]
        if truckSize >= boxes:
          total += boxes*units
          truckSize -= boxes
        elif 0 < truckSize < boxes:
          total += truckSize*units
          truckSize = 0
        else:
          break
      
      return total